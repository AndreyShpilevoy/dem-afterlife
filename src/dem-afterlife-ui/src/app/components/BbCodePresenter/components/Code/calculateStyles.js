const calculateStyles = themeName => ({
    code: {
        backgroundColor: '#59534D',
        border: '0.0625rem solid #9C9085',
        padding: 0.4
    },
    codeHeader: {
        color: '#D4D4D4',
        display: 'flex'
    },
    codeHeaderLeft: {
        marginRight: 'auto'
    },
    codeHeaderRight: {
        marginLeft: 'auto'
    },
    codeContent: {
        color: '#FFA100',
        fontFamily: 'monospace',
        whiteSpace: 'pre-wrap'
    },
    codeSelectButton: {
        color: '#CBC065',
        cursor: 'pointer',
        '&:hover': {
            color: '#FFF495'
        }
    },
    options: {meta: 'Code', themeName}
});

export default calculateStyles;