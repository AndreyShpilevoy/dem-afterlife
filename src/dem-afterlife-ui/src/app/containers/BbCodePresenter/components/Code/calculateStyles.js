const calculateStyles = ({themeName}) => ({
    code: {
        backgroundColor: '#59534D',
        border: '0.0625rem solid #44423f',
        padding: 0.4
    },
    codeHeader: {
        color: '#D4D4D4',
        display: 'flex'
    },
    codeHeaderLeft: {
        marginRight: 'auto'
    },
    codeHeaderRight: {
        marginLeft: 'auto'
    },
    codeContent: {
        color: '#FFA100',
        fontFamily: 'monospace',
        whiteSpace: 'pre-wrap'
    },
    options: {meta: 'Code', themeName}
});

export default calculateStyles;