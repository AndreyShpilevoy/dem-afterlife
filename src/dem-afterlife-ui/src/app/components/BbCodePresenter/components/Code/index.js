/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-mutation: 0, fp/no-this: 0*/

import React, {PureComponent} from 'react';
import {node, string, number, shape} from 'prop-types';
import Term from 'containers/Term';
import {withStyles} from 'styles';
import calculateStyles from './calculateStyles';

// import styles from './index.scss';
// import SelectTextFromContainerById from 'services/domScripts/SelectTextFromContainerById';

class Code extends PureComponent {
    static propTypes = {
        styles: shape().isRequired,
        id: number.isRequired,
        children: node,
        options: string
    }

    // state = {
    //     buttonId: `button-${this.props.id}`,
    //     containedId: `container-${this.props.id}`
    // }

    // componentDidMount() {
    //     const {buttonId, containedId} = this.state;

        // SelectTextFromContainerById.init(buttonId, containedId);
    // }

    render() {
        const {buttonId, containedId} = this.state;
        const {children, options, styles} = this.props;
        const {code, codeHeader, codeHeaderLeft, codeSelectButton, codeHeaderRight, codeContent} = styles;
        return (
            <div className = {code}>
                <div className = {codeHeader}>
                    <div className = {codeHeaderLeft}>
                        <Term spaceAfter term={{id: 34, value: 'Code:'}} />
                        <span id={buttonId} className = {codeSelectButton}>
                            <Term spaceAfter term={{id: 35, value: 'Ctrl+A, Ctrl+C'}} />
                        </span>
                    </div>
                    <div className = {codeHeaderRight}>
                        {options || null}
                    </div>
                </div>
                <span id={containedId} className = {codeContent}>
                    {children}
                </span>
            </div>
        );
    }
}

export default withStyles(theme => calculateStyles(theme), {pureComponent: true})(Code);
