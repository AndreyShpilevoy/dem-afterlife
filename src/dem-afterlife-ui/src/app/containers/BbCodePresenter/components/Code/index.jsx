import React from 'react';
import {node, string, shape} from 'prop-types';
import Term from 'containers/Term';
import {injectSheet} from 'styles';
import calculateStyles from './calculateStyles';

const codeTerm = {id: 34, value: 'Code'};

export const CodePure = ({children, options, classes}) => {
    const {main, header, headerLeft, headerRight, content} = classes;
    return (
        <div className={main}>
            <div className={header}>
                <div className={headerLeft}>
                    <Term spaceAfter term={codeTerm} untermedPostfix=':' />
                </div>
                <div className={headerRight}>
                    {options}
                </div>
            </div>
            <span className={content}>
                {children}
            </span>
        </div>
    );
};

CodePure.propTypes = {
    classes: shape().isRequired,
    children: node.isRequired,
    options: string
};

CodePure.defaultProps = {
    options: ''
};

export default injectSheet(calculateStyles, {componentName: 'Code'})(CodePure);
