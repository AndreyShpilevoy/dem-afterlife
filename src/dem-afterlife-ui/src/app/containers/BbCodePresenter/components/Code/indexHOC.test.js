/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0, react/jsx-filename-extension:0 */

import React from 'react';
import {shallow} from 'enzyme';
import * as ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import Code from './index';

ThemedStyleSheet.default.get = () => {};

describe('Code HOC', () => {
    it('component with correct message should match expected snapshot', () => {
        expect(shallow(<Code>{'test'}</Code>)).toMatchSnapshot();
    });
});
