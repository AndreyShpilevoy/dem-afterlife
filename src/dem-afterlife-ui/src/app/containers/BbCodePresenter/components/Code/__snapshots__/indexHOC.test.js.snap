// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code HOC component with correct message should match expected snapshot 1`] = `
<Code
  options=""
  styles={
    Object {
      "content": "default-Code-content",
      "header": "default-Code-header",
      "headerLeft": "default-Code-headerLeft",
      "headerRight": "default-Code-headerRight",
      "main": "default-Code-main",
    }
  }
>
  test
</Code>
`;
